homeassistant/components/mazda/climate.py

```diff
@@ -0,0 +1,187 @@
+PRESET_DEFROSTER_FRONT_AND_REAR = "Front and Rear Defroster"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the climate platform."""
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    client = entry_data[DATA_CLIENT]
+    coordinator = entry_data[DATA_COORDINATOR]
+    region = entry_data[DATA_REGION]
+
+    async_add_entities(
+        MazdaClimateEntity(client, coordinator, index, region)
+        for index, data in enumerate(coordinator.data)
+        if data["isElectric"]
+    )
+
+
+class MazdaClimateEntity(MazdaEntity, ClimateEntity):
+    """Class for the climate entity."""
+
+    _attr_name = "Climate"
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_preset_modes = [
+        PRESET_DEFROSTER_OFF,
+        PRESET_DEFROSTER_FRONT,
+        PRESET_DEFROSTER_REAR,
+        PRESET_DEFROSTER_FRONT_AND_REAR,
+    ]
+
+    def __init__(
+        self,
+        client: MazdaAPIClient,
+        coordinator: DataUpdateCoordinator,
+        index: int,
+        region: str,
+    ) -> None:
+        """Initialize Mazda climate entity."""
+        super().__init__(client, coordinator, index)
+
+        self.region = region
+        self._attr_unique_id = self.vin
+
+        if self.data["hvacSetting"]["temperatureUnit"] == "F":
+            self._attr_precision = PRECISION_WHOLE
+            self._attr_temperature_unit = TEMP_FAHRENHEIT
+            self._attr_min_temp = 61.0
+            self._attr_max_temp = 83.0
+        else:
+            self._attr_precision = PRECISION_HALVES
+            self._attr_temperature_unit = TEMP_CELSIUS
+            if region == "MJO":
+                self._attr_min_temp = 18.5
+                self._attr_max_temp = 31.5
+            else:
+                self._attr_min_temp = 15.5
+                self._attr_max_temp = 28.5
+
+        self._update_state_attributes()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Update attributes when the coordinator data updates."""
+        self._update_state_attributes()
+
+        super()._handle_coordinator_update()
+
+    def _update_state_attributes(self) -> None:
+        # Update the HVAC mode
+        hvac_on = self.client.get_assumed_hvac_mode(self.vehicle_id)
+        self._attr_hvac_mode = HVACMode.HEAT_COOL if hvac_on else HVACMode.OFF
+
+        # Update the target temperature
+        hvac_setting = self.client.get_assumed_hvac_setting(self.vehicle_id)
+        self._attr_target_temperature = hvac_setting.get("temperature")
+
+        # Update the current temperature
+        current_temperature_celsius = self.data["evStatus"]["hvacInfo"][
+            "interiorTemperatureCelsius"
+        ]
+        if self.data["hvacSetting"]["temperatureUnit"] == "F":
```

Comment:
Do you need this conversion? You have `self._attr_temperature_unit = TEMP_FAHRENHEIT` set above in this case